File: 800/339A_HelpfulMaths.java
import java.util.*;

class Solution {

    static String get(int times, String that) {
        String s = "";
        for (int i = 0; i < times; i++) {
           s += that + "+";
        }
        return s;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int one = 0, two = 0, three = 0;
        String[] s = sc.next().split("\\+");
        sc.close();
        for (int i = 0; i < s.length; i++) {
            switch(s[i]) {
                case "1":
                    one++;
                    break;
                case "2":
                    two++;
                    break;
                case "3":
                    three++;
                    break;
            }
        }

        String out = "";
        out += get(one, "1");
        out += get(two, "2");
        out += get(three, "3");

        System.out.println(out.substring(0, out.length() - 1));
    }
}

File: 800/cpp/339A_HelpfulMaths.cpp
#include <bits/stdc++.h>

using namespace std;

string printNums(string n, int len) {
	string s = "";
	for(int i = 0; i < len; i++) {
		s += n + "+";
	}
	return s;
}

int main() {
	string s;
	cin >> s;
	int one = 0, two = 0, three = 0;
	for (int i = 0; i < s.length(); i++) {
		if (s.at(i) == '1') {
			one++;
		} else if (s.at(i) == '2') {
			two++;
		} else if (s.at(i) == '3'){
			three++;
		}
	}
	string out = "";
	out += printNums("1", one);
	out += printNums("2", two);
	out += printNums("3", three);
	cout << out.substr(0, out.length()-1) << "\n";
}